app.js

```diff
@@ -1,16 +1,35 @@
 const express = require("express");
 const app = express();
 
+app.use(express.json());
+
 const { getTopics } = require("./controllers/controllers.topics");
-const { getArticles } = require("./controllers/controllers.articles");
-const { handle500s, handle404s } = require("./controllers/errors");
+const {
+  getArticles,
+  getArticleById,
+} = require("./controllers/controllers.articles");
+const {
+  handle500s,
+  handle404s,
+  handle400s,
+  handleCustomErrors,
+} = require("./controllers/errors");
 
 app.get("/api/topics", getTopics);
 
 app.get("/api/articles", getArticles);
 
-app.all("*", handle404s);
+app.get("/api/articles/:article_id", getArticleById);
+
+app.all("*", handleCustomErrors);
+
+app.use(handle500s, (err, req, res, next) => {
+  console.log(err);
+  next(err);
```

Comment:
You need to rearrange your error handlers slightly, the 500 handler should be last and should include `res.status(500).send({ msg: "internal server error" }) - this is because the 500 handler is your last line of defence, if an error makes it here it should be because it was unexpected ( such as an error in your code )

Reply:
thanks!